projects/services/consensus-service-typescript/src/consensus-service.ts

```diff
@@ -15,21 +15,21 @@ export default class ConsensusService extends Service {
   private virtualMachine: types.VirtualMachineClient;
   private blockStorage: types.BlockStorageClient;
 
-  public constructor() {
-    super();
+  public constructor(gossip: types.GossipClient, virtualMachine: types.VirtualMachineClient, blockStorage: types.BlockStorageClient, serviceConfig: ServiceConfig) {
+    super(serviceConfig);
+    this.gossip = gossip;
```

Comment:
Just a side note: you can use a member assignment shortcut. 

Instead of:

```ts
private gossip: types.GossipClient;

public constructor(gossip: types.GossipClient ...
```

You can achieve the same via:

```ts
// You don't need me now! private gossip: types.GossipClient;

public constructor(private gossip: types.GossipClient ...
```